modules/payment/src/Plugin/Commerce/CheckoutPane/PaymentInformation.php

```diff
@@ -317,6 +317,7 @@ public function validatePaneForm(array &$pane_form, FormStateInterface $form_sta
    * {@inheritdoc}
    */
   public function submitPaneForm(array &$pane_form, FormStateInterface $form_state, array &$complete_form) {
+    $this->order = $this->checkoutFlow->getOrder();
```

Comment:
@bojanz this fixes it. Along with rebuild refresh of order. So.. uh. I'm not sure what to do. Do we have the base form always refresh the order object? I can't think of how we can work around this.